src/provider/controller/ExpiryController.ts

```diff
@@ -0,0 +1,112 @@
+import {injectable, inject} from 'inversify';
+
+import {Store} from '../store/Store';
+import {Inject} from '../common/Injectables';
+import {StoredNotification} from '../model/StoredNotification';
+import {RemoveNotifications, RootAction, CreateNotification, ExpireNotification} from '../store/Actions';
+import {RootState} from '../store/State';
+
+import {AsyncInit} from './AsyncInit';
+
+@injectable()
+export class ExpiryController extends AsyncInit {
+    private readonly _store: Store<RootState, RootAction>;
+
+    private _nextExpiry: {
+        note: StoredNotification,
+        timerHandler: number
+    } | null = null;
+
+    public constructor(@inject(Inject.STORE) store: Store<RootState, RootAction>) {
+        super();
+
+        this._store = store;
+        this._store.onAction.add(this.onAction, this);
```

Comment:
is there any significance for the order of these callback executions throughout the provider? (a heads-up, now we dispatch RemoveNotification in index.ts's onAction method directly after an `notification-action`) would it be safer and/or cleaner to invoke this from provider:index.ts:onAction and keep this class as a non-AsyncInit and non-injectable instance owned by index as it has no interaction with Database or apiHandler and the design dictates this class to be a single-instance sub-module.

Reply:
Leaving as discussed